src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb.model.ts

```diff
@@ -0,0 +1,14 @@
+export interface SMBCluster {
+  cluster_id: string;
+  auth_mode: string;
+  intent: string;
+  domain_settings?: DomainSettings;
+  user_group_settings?: string[];
+  custom_dns: string[];
+  placement: string;
```

Comment:
Following: https://docs.ceph.com/en/latest/mgr/smb/#cluster-resource
Maybe we should also add `clustering` and `public_addrs` even if we do not really list this fields for now as they are only meant for clustering.

Also the `custom_dns`, `placement`,  `clustering` and `public_addrs` fields should be optional as per the documentation

Reply:
Yup correct just added.